### ISSUE TITLE
Missing secure AUTH methods

### ISSUE THREAD

**Issue by viossat** [tomav/docker-mailserver#309]
Hi,

By default, there is no available secure AUTH method:
```
250-AUTH PLAIN LOGIN
250-AUTH=PLAIN LOGIN
```

I think methods such as `CRAM-MD5` should be available instead of this image supports TLS/STARTTLS.

**Comment by Josef-Friedrich** [comment]
Can you help to implement more secure AUTH methods and write tests?

**Comment by viossat** [comment]
Sure. But in the meantime I discovered that raises another issue:
```
Requested CRAM-MD5 scheme, but we have only SHA512-CRYPT
```
Indeed, [enable non-plaintext authentication requires to store passwords as plaintext](http://wiki.dovecot.org/Authentication/Mechanisms#Non-plaintext_authentication).

I currently hash the passwords with `SHA512-CRYPT`. And in my case, I don’t want to allow admins see the password of all users.

According to [RFC 2487](https://tools.ietf.org/html/rfc2487) I mustn’t force TLS encryption (public server).

I just tried to add an account in Thunderbird without any encryption configuration. It shown a big security warning. I hope this is the case of most clients.
